src/VisualStudio/Core/Def/SampleToolWindow/SampleToolboxUserControl.xaml.cs

```diff
@@ -0,0 +1,460 @@
+
+                // TODO: proper workaround such that context.ClientCapabilities?.TextDocument?.DocumentSymbol?.HierarchicalDocumentSymbolSupport == true
+                var response = await lspService.RequestAsync(
+                    textBuffer: textBuffer,
+                    method: method,
+                    capabilitiesFilter: capabilitiesFilter,
+                    languageServerName: languageServerName,
+                    parameterFactory: ParameterFactory,
+                    cancellationToken: cancellationToken).ConfigureAwait(false);
+
+                if (response is not null && response.Response is not null)
+                {
+                    await threadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
+                    var body = response.Response.ToObject<DocumentSymbol[]>();
+                    var docSymbols = DocumentOutlineHelper.GetDocumentSymbols(body);
+                    this.originalTree = docSymbols;
+                    symbolTree.ItemsSource = docSymbols;
+                }
+                else
+                {
+                    symbolTree.ItemsSource = new List<DocSymbol>();
+                }
+            }
+        }
+
+        private void ExpandAll(object sender, RoutedEventArgs e)
+        {
+            var documentSymbols = new List<DocSymbol>();
+            foreach (var item in (List<DocSymbol>)symbolTree.ItemsSource)
+            {
+                documentSymbols.Add(ExpandAllNodes(item));
+            }
+
+            symbolTree.ItemsSource = documentSymbols;
+        }
+
+        private DocSymbol ExpandAllNodes(DocSymbol treeItem)
+        {
+            treeItem.IsExpanded = true;
+            foreach (var childItem in treeItem.Children.OfType<DocSymbol>())
+            {
+                ExpandAllNodes(childItem);
+            }
+
+            return treeItem;
+        }
+
+        private void CollapseAll(object sender, RoutedEventArgs e)
+        {
+            var documentSymbols = new List<DocSymbol>();
+            foreach (var item in (List<DocSymbol>)symbolTree.ItemsSource)
+            {
+                documentSymbols.Add(CollapseAllNodes(item));
+            }
+
+            symbolTree.ItemsSource = documentSymbols;
+        }
+
+        private DocSymbol CollapseAllNodes(DocSymbol treeItem)
+        {
+            treeItem.IsExpanded = false;
+            foreach (var childItem in treeItem.Children.OfType<DocSymbol>())
+            {
+                CollapseAllNodes(childItem);
+            }
+
+            return treeItem;
+        }
+
+        private void RemoveText(object sender, EventArgs e)
+        {
+            if (searchBox.Text == "Search Document Outline")
+            {
+                searchBox.Text = "";
+            }
+        }
+
+        private void AddText(object sender, EventArgs e)
+        {
+            if (string.IsNullOrWhiteSpace(searchBox.Text))
+                searchBox.Text = "Search Document Outline";
```

Comment:
We want all the text we show to developers in the IDE to be localized their own language. To do this we place text in Resx files and we have a system that ensures that we load and use the right (human) language.

In your case you want to add these new entries to  the [ServicesVSResources.resx](https://github.com/dotnet/roslyn/blob/main/src/VisualStudio/Core/Def/ServicesVSResources.resx).

1. Open [ServicesVSResources.resx](https://github.com/dotnet/roslyn/blob/main/src/VisualStudio/Core/Def/ServicesVSResources.resx) in Visual Studio
2. Add a new entry named `Search_Document_Outline` with the text `Search Document Outline`
3. Replace the string with `ServicesVSResources.Search_Document_Outline` in your code